cloudsync/sync/manager.py

```diff
@@ -322,6 +322,24 @@ def sync(self, sync: SyncEntry) -> bool:  # pylint: disable=too-many-branches,to
             self.finished(REMOTE, sync)
             return True
 
+        if self.in_backoff and sync.is_latest():
+            # Sync manager is in backoff (degraded performance) and the following get_latest() call will not make any
+            # provider calls -- make dummy calls to force providers to reconnect and fetch something.
+            #
+            # If successful, then it is safe to clear backoff even though this iteration will not do any syncing.
+            # If this fails then an exception is raised and backoff is increased.
+            #
+            # The idea is to resolve the degraded performance that comes with operating with a backoff timeout
+            # as quickly as possible, rather than wait for it to resolve itself "naturally".
+            #
+            # TODO: fix this -- is_latest() returns False when only LOCAL changed, so this block is skipped, but the
+            #                   get_latest() call below will not hit REMOTE, and backoff is cleared anyway
+            self.providers[LOCAL].reconnect()
```

Comment:
will exceptions here work?

Reply:
You're probably right, reconnect() here is kinda redundant (event manager already handles reconnects in its own thread).

I don't see anything to prevent sync() from being called while disconnected though, reconnecting here could save an "in backoff" sync manager cycle in some cases. But I'm not married to it.

Reply:
addressed this, as well as the TODO.